MovieBooking/Modules/Authentication/ViewModel/AuthenticationViewModel.swift

```diff
@@ -0,0 +1,74 @@
+import UIKit
+import FirebaseAuth
+import FirebaseFirestore
+
+class AuthenticationViewModel {
+    
+    // MARK: - Variables
+    var logInValidationError: ((String) -> Void)?
+    var signUpValidationError: ((String) -> Void)?
+    var onLogInSuccess: ((String) -> Void)?
+    var onSignUpSuccess: ((String) -> Void)?
+    
+    // MARK: - File Private Functions
+    fileprivate func validateFieldsforLogIn(_ email: String, _ password: String) -> String? {
+        if email.isEmpty {
+            return "Email is Empty"
+        } else if password.isEmpty {
+            return "Password is Empty"
+        } else if password.count < 8 {
+            return "Need atleast 8 characters"
+        } else if !email.contains("@") || !email.contains(".") {
+            return "Wrong email"
+        }
+        return nil
+    }
+    
+    func doLogIn(_ email: String, _ password: String) {
+        let validationResult = validateFieldsforLogIn(email, password)
+        if let validationResult = validationResult {
+            logInValidationError?(validationResult)
+        } else {
+            Auth.auth().signIn(withEmail: email, password: password) { [weak self] authResult, error in
+                guard self != nil else { return }
+                if let _ = error {
+                    self?.logInValidationError?("The user may have been deleted")
+                } else {
+                    self?.onLogInSuccess?("Success Log In")
+                }
+            }
+        }
+    }
+    
+    func doSignUp(_ email: String, _ password: String, _ firstName: String, _ lastName: String) {
+        let validationResult = validateSignUp(email, password, firstName, lastName)
+        
+        if let validationResult = validationResult {
+            signUpValidationError?(validationResult)
+        } else {
+            Auth.auth().createUser(withEmail: email, password: password) { result, error in
+                if error != nil {
+                    self.signUpValidationError?("Error in SignUp")
+                } else {
+                    let db = Firestore.firestore()
```

Comment:
you can create this variable in constant file so you can use it where ever you need